### PR TITLE
Compiler option to print leading space for `print *`

### LINKED ISSUE #1744: Add a compiler option to make `print *, "ok"` print a leading space
See https://fortran-lang.discourse.group/t/why-does-print-output-a-leading-space/5903 for the reasoning.

This should also be the default when standard conformance is requested, see https://github.com/lfortran/lfortran/issues/1677.

### PR DESCRIPTION
**Pull request by Shaikh-Ubaid**
closes #1744

### CODE REVIEW

**Comment by Shaikh-Ubaid** [comment]
Example:
```bash
$ cat integration_tests/print_04.f90 
program print_04
    implicit none
    integer :: x

    x = 24

    print *, "x is ", x
    write (*, *) "x is ", x
    print *, "ok"
    write (*, *) "ok"

 end program
$ gfortran integration_tests/print_04.f90 && ./a.out  
 x is           24
 x is           24
 ok
 ok
$ lfortran integration_tests/print_04.f90                
x is  24
x is  24
ok
ok
$ lfortran integration_tests/print_04.f90 --leading-space
 x is  24
 x is  24
 ok
 ok
```

File: src/bin/lfortran.cpp [hunk 1]
```diff
@@ -1709,6 +1709,7 @@ int main(int argc, char *argv[])
         app.add_flag("--print-targets", print_targets, "Print the registered targets");
         app.add_flag("--implicit-typing", compiler_options.implicit_typing, "Allow implicit typing");
         app.add_flag("--implicit-interface", compiler_options.implicit_interface, "Allow implicit interface");
+        app.add_flag("--leading-space", compiler_options.leading_space, "Print leading white space for list-directed format");
```
Comment (certik) [inline src/bin/lfortran.cpp:1712]: How about this:
```suggestion
        app.add_flag("--print-leading-space", compiler_options.leading_space, "Print leading white space for `print *`");
```

File: src/lfortran/semantics/ast_body_visitor.cpp [hunk 2]
```diff
@@ -2327,16 +2335,27 @@ class BodyVisitor : public CommonVisitor<BodyVisitor> {
...
         if (x.m_fmt != nullptr) {
             this->visit_expr(*x.m_fmt);
             fmt = ASRUtils::EXPR(tmp);
+        } else {
+            if (compiler_options.leading_space) {
+                ASR::ttype_t *str_type_len_0 = ASRUtils::TYPE(ASR::make_Character_t(
+                al, x.base.base.loc, 1, 0, nullptr, nullptr, 0));
+                ASR::expr_t *empty_string = ASRUtils::EXPR(ASR::make_StringConstant_t(
+                al, x.base.base.loc, s2c(al, ""), str_type_len_0));
+                body.push_back(al, empty_string);
```
Comment (certik) [inline src/lfortran/semantics/ast_body_visitor.cpp:2349]: Wouldn't this add more than 1 space? I think we'll have to handle this in the backend, especially with regards to #1749.

Comment (Shaikh-Ubaid) [reply src/lfortran/semantics/ast_body_visitor.cpp:2349]: It currently prints a single space. 

We provided an empty string to print. The space is being printed by the default separator (which is space; it seems in LFortran, we do not yet support providing a custom/user-supplied separator to print) between two arguments of print.

Comment (certik) [reply src/lfortran/semantics/ast_body_visitor.cpp:2349]: I see. It's probably fine for now.

**Comment by certik** [comment]
Let's resolve conflicts and change the compiler option name. This one is hard to test, we can test it in tests/ and test the ASR.

**Review by certik — APPROVED** [review]
Looks great!

**Pull request merged**